src/modm/platform/timer/stm32/advanced.hpp.in

```diff
@@ -199,16 +207,16 @@ public:
 	}
 
 	static inline void
-	applyAndReset()
+	generateEvent(Event ev)
 	{
-		// Generate Update Event to apply the new settings for ARR
-		generateEvent(Event::Update);
+		TIM{{ id }}->EGR |= static_cast<uint32_t>(ev);
```

Comment:
According to the reference manual the `EGR` register is write-only. We should probably have a plain assignment here instead of `|=`,